Sources/MapboxCoreNavigation/NativeHandlersFactory.swift

```diff
@@ -87,10 +92,13 @@ class NativeHandlersFactory {
             ]
         ]
         
-        var customConfig = ""
-        if let jsonDataConfig = try? JSONSerialization.data(withJSONObject: historyAutorecordingConfig, options: []),
+        var customConfig = UserDefaults.standard.dictionary(forKey: customConfigKey) ?? [:]
+        customConfig.deepMerge(with: historyAutorecordingConfig)
+        
+        var customConfigJSON = ""
+        if let jsonDataConfig = try? JSONSerialization.data(withJSONObject: customConfig, options: []),
            let encodedConfig = String(data: jsonDataConfig, encoding: .utf8) {
-            customConfig = encodedConfig
+            customConfigJSON = encodedConfig
         }
```

Comment:
[Optional] You can keep `customConfigJSON` as constant by:
```swift
let customConfigJSON: String
if let jsonDataConfig = try? JSONSerialization.data(withJSONObject: customConfig, options: []),
  let encodedConfig = String(data: jsonDataConfig, encoding: .utf8) {
  customConfigJSON = encodedConfig
}
else {
  customConfigJSON = ""
}
```

This way your clearly convey that this property won't change underneath anywhere else (helps to read code).